lgc/patch/PatchInOutImportExport.cpp

```diff
@@ -4568,18 +4484,38 @@ void PatchInOutImportExport::createTessBufferStoreFunction() {
   Value *tfBufferOffset = BinaryOperator::CreateMul(relPatchId, tfByteStride, "", tfStoreBlock);
 
   tfBufferOffset = BinaryOperator::CreateAdd(tfBufferOffset, tfByteOffset, "", tfStoreBlock);
-  tfBufferOffset = BinaryOperator::CreateAdd(tfBufferOffset, tfBufferBase, "", tfStoreBlock);
 
   auto branch = BranchInst::Create(endBlock, tfStoreBlock);
 
+  // Create llvm.amdgcn.raw.tbuffer.store.
+  std::vector<unsigned> formats;
+  if (m_gfxIp.major <= 9) {
+    formats = {
+        ((BUF_NUM_FORMAT_FLOAT << 4) | (BUF_DATA_FORMAT_32)),
+        ((BUF_NUM_FORMAT_FLOAT << 4) | (BUF_DATA_FORMAT_32_32)),
+        ((BUF_NUM_FORMAT_FLOAT << 4) | (BUF_DATA_FORMAT_32_32_32)),
+        ((BUF_NUM_FORMAT_FLOAT << 4) | (BUF_DATA_FORMAT_32_32_32_32)),
+    };
+  } else if (m_gfxIp.major == 10) {
+    formats = {BUF_FORMAT_32_FLOAT, BUF_FORMAT_32_32_FLOAT, BUF_FORMAT_32_32_32_FLOAT, BUF_FORMAT_32_32_32_32_FLOAT};
+  } else
+    llvm_unreachable("Not implemented!");
```

Comment:
nit: missing braces around the `else` body

Reply:
Fixed.